script_translator.cc无法编译

librime-1.7.2.tar.gz

[ 62%] Building CXX object src/CMakeFiles/rime.dir/rime/gear/script_translator.cc.o
/home/xxxx/temp/rime-builds/librime-1.7.2/src/rime/gear/script_translator.cc: 在成员函数‘void rime::ScriptTranslation::PrepareCandidate()’中:
/home/xxxx/temp/rime-builds/librime-1.7.2/src/rime/gear/script_translator.cc:490:23: 错误：‘exp’在此作用域中尚未声明
  490 |     cand->set_quality(exp(entry->weight) +
      |                       ^~~
/home/xxxx/temp/rime-builds/librime-1.7.2/src/rime/gear/script_translator.cc:504:23: 错误：‘exp’在此作用域中尚未声明
  504 |     cand->set_quality(exp(entry->weight) +
      |                       ^~~
make[2]: *** [src/CMakeFiles/rime.dir/build.make:1238：src/CMakeFiles/rime.dir/rime/gear/script_translator.cc.o] 错误 1
make[1]: *** [CMakeFiles/Makefile2:233：src/CMakeFiles/rime.dir/all] 错误 2
make: *** [Makefile:146：all] 错误 2

搜索了之后发现似乎和编译时的条件 -lm 有关，但不知如何处理。

--- a/src/rime/gear/script_translator.cc
+++ b/src/rime/gear/script_translator.cc
@@ -8,6 +8,7 @@
 //
 #include <algorithm>
 #include <stack>
+#include <cmath>
 #include <boost/algorithm/string/join.hpp>
 #include <boost/range/adaptor/reversed.hpp>
 #include <rime/composition.h>

--- a/src/rime/gear/table_translator.cc
+++ b/src/rime/gear/table_translator.cc
@@ -6,6 +6,7 @@
 //
 #include <boost/algorithm/string.hpp>
 #include <boost/range/adaptor/reversed.hpp>
+#include <cmath>
 #include <utf8.h>
 #include <rime/candidate.h>
 #include <rime/common.h>

